src/common.h

```diff
@@ -236,37 +347,176 @@ extern void *mallocz_int(const char *file, const char *function, const unsigned
 extern void *reallocz_int(const char *file, const char *function, const unsigned long line, void *ptr, size_t size);
 extern void freez_int(const char *file, const char *function, const unsigned long line, void *ptr);
 #else
+/**
+ * `strdup()` with failure handling.
+ *
+ * Exit netdata on failure.
+ *
+ * If NETDATA_LOG_ALLOCATIONS is set, this is logged.
+ *
+ * @see man 3 strdup
+ *
+ * @param s String to copy.
+ * @return new string
+ */
 extern char *strdupz(const char *s) MALLOCLIKE NEVERNULL;
+/**
+ * `calloc()` with failure handling.
+ *
+ * Exit netdata on failure.
+ *
+ * If NETDATA_LOG_ALLOCATIONS is set, this is logged.
+ *
+ * @see man 3 malloc
+ *
+ * @param nmemb number of objects that are `size` bytes to allocate memory for
+ * @param size number of bytes of one object
+ * @return Pointer to allocated memory
+ */
 extern void *callocz(size_t nmemb, size_t size) MALLOCLIKE NEVERNULL;
+/**
+ * `malloc()` with failure handling.
+ *
+ * Exit netdata on failure.
+ *
+ * If NETDATA_LOG_ALLOCATIONS is set, this is logged.
+ *
+ * @see man 3 malloc
+ *
+ * @param size Number of bytes to allocate.
+ * @return Pointer to allocated memory
+ */
 extern void *mallocz(size_t size) MALLOCLIKE NEVERNULL;
+/**
+ * `realloc()` with failure handling.
+ *
+ * Exit netdata on failure.
+ *
+ * If NETDATA_LOG_ALLOCATIONS is set, this is logged.
+ *
+ * @see man 3 malloc
+ *
+ * @param ptr Memory to reallocate
+ * @param size Number of bytes to reallocate.
+ * @return Pointer to allocated memory.
+ */
 extern void *reallocz(void *ptr, size_t size) MALLOCLIKE NEVERNULL;
+/**
+ * Calls `free()`.
+ *
+ * If NETDATA_LOG_ALLOCATIONS is set, this is logged.
+ *
+ * @see man 3 malloc
+ *
+ * @param ptr Pointer to allocated memory.
+ */
 extern void freez(void *ptr);
 #endif
 
+/**
+ * \todo ktsaou: your help needed. Especially when this should be used.
+ *
+ * @param dst
+ * @param src
+ * @param size
+ */
 extern void json_escape_string(char *dst, const char *src, size_t size);
 
+/**
+ * \todo ktsaou: your help needed. Especially what is the difference to mmap 
```

Comment:
1. it creates the file if does not exist (to the correct size).
2. it truncates the file if it already exists and is bigger than expected.
3. sets various memory management heuristics we know netdata memory databases use (like `MADV_SEQUENTIAL`)
4. prevents this memory from being copied when netdata forks to execute its plugins.
5. enables KSM for it.
6. in memory mode `ram`, it loads the file into memory (in all other modes, loading it is on-demand by the kernel).